middleware/rewrite/class.go

```diff
@@ -1,26 +1,33 @@
-// Package rewrite is middleware for rewriting requests internally to something different.
 package rewrite
 
 import (
+	"fmt"
 	"strings"
 
 	"github.com/miekg/dns"
 )
 
-// ClassRule is a class rewrite rule.
-type ClassRule struct {
+type classRule struct {
 	fromClass, toClass uint16
 }
 
-// New initializes rule.
-func (rule ClassRule) New(args ...string) Rule {
-	from, to := args[0], strings.Join(args[1:], " ")
-	return &ClassRule{dns.StringToClass[from], dns.StringToClass[to]}
-
+func newClassRule(args ...string) (Rule, error) {
```

Comment:
don't fully understand this, varargs but then check for enough params? (it's also internal, so it's not part of any API)